register and login linked to backend functions

davidli731 opened the pull request:
login and register functional with the backend. Still need to add admin registration and login and pop up notification for invalid inputs.

davidli731 committed:
register and login linked to backend functions

login and register functional with the backend.

jasmine-nahrain commented on src/users/register.js:
```diff
@@ -79,12 +83,29 @@ export default class Register extends Component {
     localStorage.setItem('student_id', this.state.studentID);
     localStorage.setItem('email', this.state.email);
     localStorage.setItem('password', this.state.password);
-    window.location.href = '/dnd';
```
need this to route user to main page

jasmine-nahrain commented on src/users/register.js:
```diff
@@ -79,12 +83,29 @@ export default class Register extends Component {
+
+
+    /*localStorage.setItem('fname', this.state.fname);
     localStorage.setItem('lname', this.state.lname);
```
You can delete this as long as its not needed

jasmine-nahrain commented on src/users/userInfo.js:
```diff
@@ -0,0 +1,30 @@
+    })
+}
+
+export const login = user => {
```
need to save to local storage aswell

jasmine-nahrain commented on src/circuit/main.js:
```diff
@@ -100,6 +105,11 @@ export default class Main extends Component {
       this.redoButton.current.disabled = (vers === finalvers);
+
+      const token = localStorage.token;
+      this.setState({
```
Cant use state in this file. its annoying but it breaks the main page

jasmine-nahrain commented on src/users/login.js:
```diff
@@ -57,10 +59,20 @@ export default class Login extends Component {
+    /*localStorage.setItem('email', this.state.email);
     localStorage.setItem('password', this.state.password);
     localStorage.setItem('loggedIn', true);
-    window.location.href = '/dnd';
```
need this to reroute user to main page or have error message show if cannot login. right now it doesn't show anything for me

davidli731 committed:
Updated login and rego functions

rego now re-routes to main page after successful registration, data is also stored to localStorage, removed state use from main (all necessary data is already stored in localStorage anyway such as email), added error popup on failed login or rego attempt.
